app/routes.py

```diff
@@ -1,2 +1,239 @@
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": bool(new_task.completed_at)}}
+        
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])   
+def get_tasks():
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks: 
+            tasks_response.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                })
+
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if not task:
+        return make_response(f"Task {task_id} not found", 404)
+        
+    if request.method == "GET":
+        if not task.goal_id:
+            response_body = {"task":
+                {"id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+        
+        else:
+            response_body = {"task":
+            {"id": task.id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        task.title = request_body['title']
+        task.description = request_body['description']
+        if 'completed_at' in request_body:
+            task.is_complete = True
+        else:
+            task.is_complete = False
+        db.session.commit()
+        
+        response_body = {"task": {"id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+        
+        return jsonify(response_body), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {"details":(f'Task {task.id} "{task.title}" successfully deleted')}, 200
+
+
+def slack_bot(title):
+    query_path = {'channel':'slack_api_test_channel', 'text': title}
+    header = {'Authorization': os.environ.get('BOT')}
+    response=requests.post('https://slack.com/api/chat.postMessage', params = query_path, headers= header)
+    return response.json()
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_completed_task(task_id):
+    task = Task.query.get_or_404(task_id)
```

Comment:
Be careful using `get_or_404`. We are writing a JSON-based RESTful web api, but without additional configuration, the error raised by `get_or_404` will result in sending back `text/html` as the result. Check out the Flask documentation for [error handlers](https://flask.palletsprojects.com/en/2.0.x/errorhandling/#error-handlers) for some ideas about what to do to make good use of `get_or_404`.